plugin.video.dailymotion_com/default.py

```diff
@@ -0,0 +1,629 @@
+    addDir(translation(30037), urlMain+"/user/"+dmUser+"/favorites?fields=description,duration,id,owner.username,taken_time,thumbnail_large_url,title,views_total&sort=recent&limit="+itemsPerPage+"&family_filter="+familyFilter+"&localization="+language+"&page=1", 'listVideos', "")
+    addDir(translation(30038), urlMain+"/user/"+dmUser+"/playlists?fields=id,name,videos_total&sort=recent&limit="+itemsPerPage+"&family_filter="+familyFilter+"&localization="+language+"&page=1", 'listUserPlaylists', "")
+    addDir(translation(30042), urlMain+"/user/"+dmUser+"/groups?fields=id,name,description&sort=recent&limit="+itemsPerPage+"&family_filter="+familyFilter+"&localization="+language+"&page=1", 'listGroups', "")
+    xbmcplugin.endOfDirectory(pluginhandle)
+
+
+def listUserPlaylists(url):
+    content = getUrl(url)
+    content = json.loads(content)
+    for item in content['list']:
+        id = item['id']
+        title = item['name'].encode('utf-8')
+        vids = item['videos_total']
+        addDir(title+" ("+str(vids)+")", urllib.quote_plus(str(id)+"_"+dmUser+"_"+title), 'showPlaylist', '')
+    if content['has_more']:
+        currentPage = content['page']
+        nextPage = currentPage+1
+        addDir(translation(30001)+" ("+str(nextPage)+")", url.replace("page="+str(currentPage), "page="+str(nextPage)), 'listUserPlaylists', "")
+    xbmcplugin.endOfDirectory(pluginhandle)
+
+
+def listGroups(url):
+    if url == "ALL":
+        url = urlMain+"/groups?fields=id,name,description&sort=recent&filters=featured&limit="+itemsPerPage+"&family_filter="+familyFilter+"&localization="+language+"&page=1"
+    content = getUrl(url)
+    content = json.loads(content)
+    for item in content['list']:
+        id = item['id']
+        title = item['name'].encode('utf-8')
+        desc = item['description'].encode('utf-8')
+        addDir(title, "group:"+id, 'sortVideos1', '', desc)
+    if content['has_more']:
+        currentPage = content['page']
+        nextPage = currentPage+1
+        addDir(translation(30001)+" ("+str(nextPage)+")", url.replace("page="+str(currentPage), "page="+str(nextPage)), 'listGroups', "")
+    xbmcplugin.endOfDirectory(pluginhandle)
+
+
+def showPlaylist(id):
+    url = urlMain+"/playlist/"+id+"/videos?fields=description,duration,id,owner.username,taken_time,thumbnail_large_url,title,views_total&sort=recent&limit="+itemsPerPage+"&family_filter="+familyFilter+"&localization="+language+"&page=1"
+    listVideos(url)
+
+
+def favouriteUsers():
+    xbmcplugin.addSortMethod(pluginhandle, xbmcplugin.SORT_METHOD_LABEL)
+    if os.path.exists(channelFavsFile):
+       with open(channelFavsFile, 'r') as fh:
+          content = fh.read()
+          match = re.compile('###USER###=(.+?)###THUMB###=(.*?)###END###', re.DOTALL).findall(content)
+          for user, thumb in match:
+            addUserFavDir(user, 'owner:'+user, 'sortVideos1', thumb)
+       fh.close()
```

Comment:
you dont need to close manually when using "with". Itś fine to fix that for next update though.